utils/providerAPI/getUserRepositories.ts

```diff
@@ -52,6 +93,42 @@ const getUserRepositoriesForGitHub = async (access_key: string, authId?: string)
 	return allGitHubRepositories;
 }
 
+export const getUserRepositoriesForBitbucket = async (access_key: string, authId?: string) => {
+	const workspacesEndPoint = '/user/permissions/workspaces';
+	const response: { data: { values: BitbucketWorkspaceObj[] }, status: number } = await axios.get(baseURL['bitbucket'] + workspacesEndPoint, {
+		headers: {
+			'Accept': 'application/json',
+			'Authorization': `Bearer ${access_key}`
+		}
+	}) // TODO: implement pagination
+	if (response.status !== 200) {
+		throw ReferenceError(`[getUserRepositories] Could not get Bitbucket workspaces of the user (auth-id: ${authId}`);
+	}
+
+	const workspaces = response.data.values.map((workspaceObj) => workspaceObj.workspace.slug);
+
+	const repositories: RepoIdentifier[] = []
+	for (const workspace of workspaces) {
+		const repositoriesEndPoint = `/repositories/${workspace}`;
+		const response: { data: { values: BitbucketRepoObj[] }, status: number } = await axios.get(baseURL['bitbucket'] + repositoriesEndPoint, {
+			headers: {
+				'Accept': 'application/json',
+				'Authorization': `Bearer ${access_key}`
+			}
+		}) // TODO: implement pagination
+		if (response.status !== 200) {
+			console.error(`[getUserRepositories] Could not get Bitbucket repositories for the workspace (${workspace}) for auth-id ${authId}`, response.data);
+		}
+		const allBitbucketRepoIdentifiers = response.data.values.map(repoObj => ({
```

Comment:
if response.status is not 200, this line will fail, as there will be nothing in response.data.values and map call will fail. Maybe use continue inside if block of line 119 or handle it in some other way

Reply:
Nice catch, fixing this.